indra_world/assembly/incremental_assembler.py

```diff
@@ -86,6 +87,70 @@ def __init__(self, prepared_stmts,
         self.belief_scorer = eidos_scorer
         self.beliefs = self.get_beliefs()
 
+    def get_curation_effects(self, curations):
+        mappings = {}
+        for stmt_hash, curation in curations.items():
+            new_hash = self.get_curation_effect(stmt_hash, curation)
+            if new_hash:
+                mappings[stmt_hash] = new_hash
+        return mappings
+
+    def get_curation_effect(self, old_hash, curation):
+        """Return changed matches hash as a result of curation."""
+        relevant_types = {'factor_polarity', 'reverse_relation',
+                          'factor_grounding'}
+        if curation['update_type'] not in relevant_types:
+            return None
+        # This should work but we don't want to error in case
+        # the hash is missing.
+        stmt = self.stmts_by_hash.get(old_hash)
+        if not stmt:
+            return None
+        # Make a deepcopy so we don't persist changes
+        stmt = copy.deepcopy(stmt)
+        # Flip the polarity
+        if curation['update_type'] == 'factor_polarity':
+            self.apply_polarity_curation(stmt, curation)
+        # Flip subject/object
+        elif curation['update_type'] == 'reverse_relation':
+            self.apply_reverse_curation(stmt, curation)
+        # Change grounding
+        elif curation['update_type'] == 'factor_grounding':
+            self.apply_grounding_curation(stmt, curation)
+
+        new_hash = stmt.get_hash(matches_fun=self.matches_fun,
+                                 refresh=True)
+        if new_hash != old_hash:
+            return new_hash
+        else:
+            return None
+
+    @staticmethod
+    def apply_polarity_curation(stmt, curation):
+        role, new_pol = parse_factor_polarity_curation(curation)
+        if role == 'subj':
+            stmt.subj.delta.polarity = new_pol
+        elif role == 'obj':
+            stmt.obj.delta.polarity = new_pol
+
+    @staticmethod
+    def apply_reverse_curation(stmt, curation):
+        tmp = stmt.subj
```

Comment:
```suggestion
stmt.subj, stmt.obj = stmt.obj, stmt.sub
```

tuple unpacking is the best

Reply:
Thanks, I'll change that!

Reply:
oops I made a typo here though, I hope you caught it